GMP seems to require m4

Elliot suspects this is a result of #7014

### Configuration Information

- Output of `chpl --version`: chpl Version 1.17.0 pre-release

I was wrong / lydia was right -- m4 is a requirement. 

I think there's 2 next steps here:
 - adjust our documentation to list m4 as a prereq (revive https://github.com/chapel-lang/chapel/pull/7696, and add m4 to the apt/yum/zypper commands)
 - Think of a better way to let users know a speculative build failed. The gmp build actually fails with a pretty clear `configure: error: No usable m4 in $PATH`, but since it's a speculative build we sweep that under the rug and users only find out at chpl compilation time with incredibly unhelpful   unknown __mpz_struct (and other gmp type) errors. Maybe we could do something like set CHPL_GMP=failed if the speculative build failed, and in the module code emit a clear error? Or we could have a "production" Chapel install that did a non-speculative build of gmp? Something to think about -- I'll probably open a new issue for this.

--- a/doc/rst/usingchapel/prereqs.rst
+++ b/doc/rst/usingchapel/prereqs.rst
@@ -33,6 +33,8 @@ about your environment for using Chapel:
       expression support (i.e.  CHPL_LLVM=llvm or CHPL_REGEXP=re2). If
       GCC is used, we recommend GCC version 5 or newer for this purpose.
 
+  * Building GMP requires an M4 macro processor
+
   * Building LLVM requires cmake version 3.4.3 or later.
 
   * If you wish to use Chapel's test system, python-setuptools and
@@ -47,20 +49,20 @@ We have used the following commands to install the above prerequisites:
 
   * CentOS, Fedora 21::
 
-      sudo yum install gcc gcc-c++ perl python python-devel python-setuptools bash make gawk
+      sudo yum install gcc gcc-c++ m4 perl python python-devel python-setuptools bash make gawk
 
   * Fedora 22::
 
-      sudo dnf install gcc gcc-c++ perl python python-devel python-setuptools bash make gawk
+      sudo dnf install gcc gcc-c++ m4 perl python python-devel python-setuptools bash make gawk
 
   * SLES, openSUSE::
 
-      sudo zypper install gcc gcc-c++ perl python python-devel python-setuptools bash make gawk
+      sudo zypper install gcc gcc-c++ m4 perl python python-devel python-setuptools bash make gawk
 
   * Debian, Ubuntu::
 
-      sudo apt-get install gcc g++ perl python python-dev python-setuptools bash make mawk
+      sudo apt-get install gcc g++ m4 perl python python-dev python-setuptools bash make mawk
 
   * FreeBSD::
 
-     sudo pkg install bash python py27-setuptools gmake gawk
+     sudo pkg install m4 python py27-setuptools bash gmake gawk

